contracts/test/integration/vip/vip13.sol

```diff
@@ -0,0 +1,477 @@
+                arguments: abi.encodeWithSignature(
+                    "connectDeposit(address,address)",
+                    address(voltV2DaiPriceBoundPSM),
+                    MainnetAddresses.COMPOUND_DAI_PCV_DEPOSIT
+                ),
+                description: "Connect DAI deposit to PSM in ERC20 Allocator"
+            })
+        );
+
+        proposal.push(
+            ITimelockSimulation.action({
+                value: 0,
+                target: MainnetAddresses.PCV_GUARDIAN,
+                arguments: abi.encodeWithSignature(
+                    "addWhitelistAddresses(address[])",
+                    toWhitelist
+                ),
+                description: "Add new DAI, and USDC PSMs to PCV Guardian whitelist"
+            })
+        );
+        prop = proposal;
+    }
+
+    function mainnetSetup() public override {
+        // new pcv guardian to be deployed with safe address as timelock
+        uint256 governorVoltBalanceBeforeUsdc = IERC20(MainnetAddresses.VOLT)
+            .balanceOf(MainnetAddresses.GOVERNOR);
+
+        vm.startPrank(MainnetAddresses.GOVERNOR);
+
+        PCVGuardian(MainnetAddresses.PCV_GUARDIAN)
+            .withdrawAllERC20ToSafeAddress(
+                MainnetAddresses.VOLT_USDC_PSM,
+                MainnetAddresses.VOLT
+            );
+
+        uint256 governorVoltBalanceAfterUsdc = IERC20(MainnetAddresses.VOLT)
+            .balanceOf(MainnetAddresses.GOVERNOR);
+
+        voltInUsdcPSM =
+            governorVoltBalanceAfterUsdc -
+            governorVoltBalanceBeforeUsdc;
+
+        uint256 governorVoltBalanceBeforeDai = IERC20(MainnetAddresses.VOLT)
+            .balanceOf(MainnetAddresses.GOVERNOR);
+
+        PCVGuardian(MainnetAddresses.PCV_GUARDIAN)
+            .withdrawAllERC20ToSafeAddress(
+                MainnetAddresses.VOLT_DAI_PSM,
+                MainnetAddresses.VOLT
+            );
+
+        uint256 governorVoltBalanceAfterDai = IERC20(MainnetAddresses.VOLT)
+            .balanceOf(MainnetAddresses.GOVERNOR);
+
+        voltInDaiPSM =
+            governorVoltBalanceAfterDai -
+            governorVoltBalanceBeforeDai;
+
+        IERC20(MainnetAddresses.VOLT).transfer(
+            MainnetAddresses.TIMELOCK_CONTROLLER,
+            voltInDaiPSM + voltInUsdcPSM
+        );
+        vm.stopPrank();
+    }
+
+    function mainnetValidate() public override {
+        // Volt Token Validations
+        assertEq(voltV2.decimals(), 18);
+        assertEq(voltV2.symbol(), "VOLT");
+        assertEq(voltV2.name(), "Volt");
+        assertEq(voltV2.totalSupply(), voltInUsdcPSM + voltInDaiPSM);
+
+        // Volt Migrator Validations
+        assertEq(address(voltMigrator.core()), MainnetAddresses.CORE);
+        assertEq(address(voltMigrator.oldVolt()), MainnetAddresses.VOLT);
+        assertEq(address(voltMigrator.newVolt()), address(voltV2));
+        assertEq(IERC20(address(voltV2)).balanceOf(address(voltMigrator)), 0);
```

Comment:
shouldn't this migrator have new Volt minted to it?

Reply:
Yes, but, we mint the amount taken from the current PSMs to the migrator, and in the same transaction we use the migrator to exchange to the new PSMs with the respective amounts taken from the old ones, so it ends up with 0 new volt at the end of the transaction